content/news/2023-06-from-single-files-to-collections/index.md

```diff
@@ -0,0 +1,180 @@
+
+</div>  
+<figcaption>
+  Proposed solution to reindexed second collection using elements of the first collection.
+</figcaption>
+</div>  
+
+However, if this logic needs to be applied to multiple collections or again for multiple steps, the workflow becomes even more unreasonably large and complex. 
+
+Furthermore it needs to be considered, that filtering empty or failed elements from a collection
+can hide the root cause of the problem. The question why some tool runs fail or produce empty output should always be further investigated.
+
+### Rerun the workflow with rerun activated 
+
+In some cases a rerun of the workflow can also solve the issue, for example in our use case where the failure was rather randomly.
+For this strategy it is beneficial to activate the rerun option of galaxy (User/Preferences/Manage Information) on instance with version > 23.0. 
+This allows to only rerun failed elements, which saves computing resources and time.
+
+<div class="center">
+<div class="img-sizer" style="width: 100%">
+
+![Empty collection](./figs/rerun.png)
+
+</div>  
+<figcaption>
+  Rerun option of galaxy using cached files.
+</figcaption>
+</div>  
+
+
+## Global tool dependent solution
+
+Since this issue cannot be solved satisfactory on the workflow level, one can still aim to improve the problem for the community by solving it on the tool level.
+In general the aim should be that the tool neither fails nor produces empty output. This is much more work, but will ultimately have a benefit for all users.
+Two things need to be considered here.
+
+* Why does the tool fail, can it be solved? In our case we had to increase the memory of kraken2 for the server. In other cases it could be necessary to inspect the tool wrapper and find individual solutions.
+* Why does the tool produce empty elements? If the tool produces empty elements when in fact it has an output but the output is an empty file (e.g. if a detection tools detects nothing), it might be as simple as adding an empty table or fasta file for this tool, to allow follow up tools to work. E.g. an assembly tool than cannot create any contig should return an empty fasta file instead of failing.
+
+# Case 3 - Collection workflow logic does not fully comply with single file logic 
+
+## Implicit Conversions
+
+Some of the tools, used in the *Foodborne pathogen detection workflow using collections*, failed when we run the workflow but succeed when the tool did run alone without a workflow, for example `Krakentools: Extract Kraken Reads By ID` and `Filter Sequence by ID`.
+
+After a bit of investigation we noticed that when these tools run alone (without being in a workflow) or in a workflow with single files galaxy performs an implicit decompressing of the zipped input files, which then channels the correct file to the underlying wrapper tool. However when the same tools run in a collection in a workflow this implicit decompressing does not take place, which cause the output to fail. 
```

Comment:
It's unclear what you're doing there. Implicit converters sure run as part of workflows. Overall I don't think this is a statement that should be published.

Reply:
Removed, but still an issue, will raise next week.